### ISSUE TITLE
autoUpdater - "The command is disabled and cannot be executed"

### ISSUE THREAD

**Issue by fancyydk** [electron/electron#7792]
- Electron version: 1.3.6
- Operating system: Mac OS X 10.10.5 and macOS 10.12

Not sure how this error occurs, but the autoUpdater downloads an update successfully then fails to update the app. Googling this error seems to have something to do with ReactiveCocoa which Squirrel.Mac depends on, but decided to start here to see if anyone's run into this issue.

**Comment by kevinsawicki** [comment]
> Electron version: 1.3.6

Are you able to reproduce this on 1.4.4 as well? Are there any errors logged to the `Console.app` when this happens?

**Comment by fancyydk** [comment]
@kevinsawicki, thanks for the quick reply.

Unfortunately, I myself cannot reproduce this error. I saw it in the logs from a couple of users and it looks like the autoUpdater sucessfully downloaded the file but that's all I can tell for now. They are all on version 1.3.6.

**Comment by salomvary** [comment]
I saw this happening when called `autoUpdater.checkForUpdates()` while it was in the middle of checking for updates. Simplest way to reproduce:

```js
autoUpdater.checkForUpdates()
autoUpdater.checkForUpdates()
```

Possible fixes on Electron:
- `autoUpdater.checkForUpdates()` is a no-op if it is already checking
- Expose a `autoUpdater.status` with meaningful value on the progress

Possible fix (or workaround) outside of Electron:
- Listen to autoUpdater events and maintain the state yourself. Do not do `autoUpdater.checkForUpdates()` if it is already happening.

**Comment by justechn** [comment]
I am also running into this and I am on electron@1.6.2

**Comment by danmo** [comment]
Has this been resolved ? Still seeing it.

**Comment by britzke** [comment]
I saw above that somebody can reproduce and someone can not. May be that is an admin/non-admin problem? For native OS X applications an authentication dialog is opened to give the ability to authenticate as admin, when trying to do admin tasks like an update in a protected directory (/Applications).
The Signal application is [struggling on an auto-update issue](https://github.com/WhisperSystems/Signal-Desktop/issues/1678), too (which is indeed an admin/non-admin issue). They are using the Electron framework and assume that the Electron framework causes the issue.

**Comment by codebytere** [comment]
Is this still occurring for folks on recent versions of electron running as admin?

**Comment by codebytere** [comment]
We are no longer implementing bugfixes for versions of Electron <= `1.7.x`, so i'm going to close this issue but if it is still persisting in more recent versions of Electron we can certainly reopen it!

**Comment by dsimic** [comment]
Hello. I am seeing the same error but on Electron v.2.0.8 and Mac OSX High Sierra (10.13.6).

**Comment by ccnokes** [comment]
Just saw this when running an update on electron 3.0.0.

**Comment by alexanderturinske** [comment]
Just saw this when running an update on electron 5.0.10

**Comment by sparky-raccoon** [comment]
Having the same issue on electron 5.0.11, and it's all about calling checkForUpdates.

**Comment by aakarim** [comment]
I have the same issue in 7.0.1 on Mac. 

I have tried using Sudo with the same results. 

The only abnormal thing about my setup is that I’m not using TLS for my update server (because Nuts doesn’t support it yet)

**Comment by callmeberzerker** [comment]
I have this same issue when trying to update Visual studio code on Catalina.

**Comment by stevenfabre** [comment]
I'm having the same issue on electron 9.0.4 on macOS Catalina 10.15.4.
To reproduce, I only need to call `autoUpdater.checkForUpdates()` while it's already checking for an update (e.g click the "Check for Updates..." menu item twice quickly or when an update is currently being downloaded.

![image](https://user-images.githubusercontent.com/2289736/90367118-b04bab80-e0ab-11ea-81c5-bdfcd4711ca3.png)

**Comment by AdrianMrn** [comment]
I'm seeing this using electron-updater 4.3.8 and electron 11.3.0, I've checked to make sure I'm not checking for updates multiple times. How did all of you manage to fix the issue?

```
09:48:41.981 › Error: The command is disabled and cannot be executed
    at u.quitAndInstall (/Users/…/app.asar/dist/main.js:113:133615)
    at /Users/…/app.asar/dist/main.js:367:24549
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

**Comment by callmeberzerker** [comment]
To whom it may concern:

After awhile of not being unable to manually updating `vscode` on my work mac machine, thx to the due diligence of one of our SysAdmins we figured out the culprit.

You can read this blog post for more info: https://soundmacguy.wordpress.com/2018/02/25/microsoft-skype-definitely-and-teams-maybe-disabling-automatic-updates/

TLDR being: If some software like SkypeForBusiness or Teams is being distributed to your work machines with the flag `DISABLE_UPDATE_CHECK 1` it will disable the updates for all apps using `Squirrel`.

**Comment by AliasT** [comment]
Solved this by use private `nativeUpdater` from [electron-builder codebase](https://github.com/electron-userland/electron-builder/blob/master/packages/electron-updater/src/MacUpdater.ts#L13), may be some other issues unkonw.
```js
const autoUpdater = require('electron-updater')
autoUpdater.nativeUpdater.quitAndInstall()
```

**Comment by mihaerzen** [comment]
@AliasT 

Well in that case you can use the **`autoUpdater`** directly from electron package as that is what is referenced in the code you've mentioned. That way you don't have to access private resources which is a dangerous pattern.

```typescript
import { autoUpdater } from 'electron';

autoUpdater.quitAndInstall();
```

**Comment by aryo** [comment]
> @AliasT
> 
> Well in that case you can use the **`autoUpdater`** directly from electron package as that is what is referenced in the code you've mentioned. That way you don't have to access private resources which is a dangerous pattern.
> 
> ```ts
> import { autoUpdater } from 'electron';
> 
> autoUpdater.quitAndInstall();
> ```

When we do this, it complains about no update being available

```
Error: No update available, can't quit and install
```

**Comment by maoweifan** [comment]
I have also encountered this issue and I have found a solution to the problem
